core_contracts/dex/dex.py

```diff
@@ -1739,3 +1789,13 @@ def addLpAddresses(self, _poolId: int, _addresses: List[Address]) -> None:
         for _address in _addresses:
             if self.balanceOf(_address, _poolId) > 0:
                 self._active_addresses[_poolId].add(_address)
+
+
+    @only_governance
+    @external
+    def repairSnapshots(self, _poolId: int, _addresses: List[Address]) -> None:
```

Comment:
Could you check the max length of _addresses to something like 100 and revert if it's bigger ? 
Considering the amount of DB writes performed in that method, if it's called with a huge address list, the transaction may timeout and create issues.